src/plugins/wizard/public/application/contributions/containers/data_tab/items/use/use_dropbox.tsx

```diff
@@ -118,12 +122,18 @@ export const useDropbox = (props: UseDropboxProps): DropboxProps => {
 
   const onDropField = useCallback(
     (data: FieldDragDataType['value']) => {
-      if (!data) return;
+      if (!data || !validAggTypes.length) {
+        throw new Error('Cannot drop field here, missing drop properties');
+      }
 
       const { name: fieldName } = data;
+      const schemaAggTypes = (schema.defaults as any).aggTypes;
+      const allowedAggTypes = schemaAggTypes
+        ? schemaAggTypes.filter((type) => validAggTypes.includes(type))
+        : [];
 
       aggConfigs?.createAggConfig({
-        type: (schema.defaults as any).aggType,
+        type: allowedAggTypes[0] || validAggTypes[0],
```

Comment:
I'm still catching up to some of the concepts in d&d, but why would we want to create an `aggConfig` even if there are no `allowedAggTypes` in the schema? Is it because schemas aren't required to define `allowedAggTypes`? If that's the case, why couldn't we just set L133 to `validAggTypes` and avoid the fallback here?

Reply:
Great question. To understand the logic here, let me back up a bit since it took me a while to figure it out myself. On app start, the agg service registers a variety of aggregation types like avg, max, etc. But each schema has filters for the agg types they support and each aggtype has filters for the field types they support. So `validAggTypes` is an array of all the agg types that the field being dragged can be used with, and if there is atleast one of them, then we should be able to drop the field into that dropbox.

`allowedAggTypes` is a subset of `validAggTypes` where the aggTypes defined in `schema.defaults` are also present in `validAggTypes`. So the logic is that we use the default aggType thats also present in  `validAggTypes`, but if that isnt possible, we just use the first agg type from `validAggTypes` as a fallback.